File: mobile/src/main/java/treehou/se/habit/ui/settings/SettingsModule.java
package treehou.se.habit.ui.settings;


import dagger.Module;
import dagger.Provides;
import treehou.se.habit.module.ViewModule;

@Module
public class SettingsModule extends ViewModule<SettingsFragment> {

    public SettingsModule(SettingsFragment fragment) {
        super(fragment);
    }

    @Provides
    public SettingsContract.View provideView() {
        return getView();
    }

    @Provides
    public SettingsContract.Presenter providePresenter(SettingsPresenter presenter) {
        return presenter;
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/settings/SettingsModule.kt
package treehou.se.habit.ui.settings


import dagger.Module
import dagger.Provides
import treehou.se.habit.module.ViewModule

@Module
class SettingsModule(fragment: SettingsFragment) : ViewModule<SettingsFragment>(fragment) {

    @Provides
    fun provideView(): SettingsContract.View {
        return view
    }

    @Provides
    fun providePresenter(presenter: SettingsPresenter): SettingsContract.Presenter {
        return presenter
    }
}
